### PR TITLE
Feat/improve space validations

### PR DESCRIPTION
**Pull request by egalvis27**
## What is Changed / Added
----
- Unified drive space validation so backups and regular drive uploads rely on the same check before uploading data.
- Updated backup space validation to reserve the actual number of bytes that must be uploaded, instead of using the net size delta between local and remote files.
- Added upload preflight validation for temporal file uploads to fail early when the account does not have enough available drive space.
- Updated the virtual drive release flow to preserve the temporal file when the upload is rejected due to insufficient space, instead of deleting the local content.
- Hardened the shared space validator to reject invalid values such as negative or non-finite requested sizes.
- Added focused test coverage for:
  - shared space validation edge cases
  - insufficient-space preflight during temporal file uploads
  - release behavior when uploads are rejected for lack of space

## Why
- Backups and normal uploads consume the same remote storage quota, so they should be validated with the same rule and produce consistent behavior.
- The previous backup validation could allow false positives by checking only the net storage delta. In practice, the flow uploads full file contents first, so the real peak space requirement is the amount being uploaded, not the final net result.
- Failing before opening streams and starting uploads avoids unnecessary work, retries, and late quota errors.
- Preserving the temporal file on insufficient-space errors prevents local data loss and makes retry behavior safer for the user.
- Centralizing this logic reduces drift between upload paths and makes future storage validation changes easier to maintain.

### CODE REVIEW

File: src/apps/backups/BackupService.ts [hunk 1]
```diff
@@ -141,15 +141,15 @@ export class BackupService {
...
+      acc += local.size;
 
       return acc;
     }, 0);
 
-    const total = bytesToUpdate + bytesToUpload;
+    const bytesToReserve = bytesToUpdate + bytesToUpload;
 
-    const validateSpaceResult = await UsageModule.validateSpace(total);
+    const validateSpaceResult = await UsageModule.validateSpace(bytesToReserve);
```
Comment (egalvis27) [inline src/apps/backups/BackupService.ts:152]: We changed this check to be simpler and safer:

Before, backup could pass the space check but still fail later during upload.  
Now, we validate space using the real bytes we are going to upload (`added + modified`).

File: src/backend/features/usage/validate-space.ts [hunk 2]
```diff
@@ -2,10 +2,14 @@ import { Result } from '../../../context/shared/domain/Result';
...
 
 /**
- * Validates if there's enough space available for the requested amount
+ * Validates if there's enough drive space available to reserve the requested amount.
  */
 export async function validateSpace(desiredSpaceToUse: number): Promise<Result<{ hasSpace: boolean }, Error>> {
   try {
+    if (!Number.isFinite(desiredSpaceToUse) || desiredSpaceToUse < 0) {
+      return { error: new Error('Desired space to use must be a finite non-negative number') };
+    }
```
Comment (egalvis27) [inline src/backend/features/usage/validate-space.ts:11]: This guard rejects invalid space requests early.  
`desiredSpaceToUse` must be a finite, non-negative number; otherwise the check could return misleading results (for example with `NaN`, `Infinity`, or negative values).  
Failing fast here makes space validation safer and easier to reason about.

File: src/backend/features/virtual-drive/services/operations/release.service.ts [hunk 3]
```diff
@@ -80,6 +81,16 @@ export async function release({ path, processName, container }: Props): Promise<
...
 
+      if (uploadError instanceof DriveDesktopError && uploadError.cause === 'NOT_ENOUGH_SPACE') {
+        logger.warn({
+          msg: '[Release] Upload preflight rejected file because drive space is insufficient, preserving temporal file without upload',
+          error: uploadError,
+          path,
+          processName,
+        });
+        return { error: new FuseIOError('Upload failed due to insufficient storage or network issues.') };
+      }
```
Comment (AlexisMora) [inline src/backend/features/virtual-drive/services/operations/release.service.ts:100]: Is an issue being pushed already in this process? otherwise it could be nice to add it

**Review by AlexisMora — APPROVED** [review]
